src/Catel.Core/Catel.Core.Shared/Logging/Listeners/FileLogListener.cs

```diff
@@ -25,19 +25,82 @@ namespace Catel.Logging
     /// </summary>
     public class FileLogListener : BatchLogListenerBase
     {
-        private const string AppData = "{AppData}";
-        private const string AppDataLocal = "{AppDataLocal}";
-        private const string AppDataRoaming = "{AppDataRoaming}";
-        private const string AppDataMachine = "{AppDataMachine}";
-        private const string AppDir = "{AppDir}";
-        private const string Date = "{Date}";
-        private const string Time = "{Time}";
-        private const string AssemblyName = "{AssemblyName}";
-        private const string AssemblyCompany = "{AssemblyCompany}";
-        private const string AssemblyProduct = "{AssemblyProduct}";
-        private const string ProcessId = "{ProcessId}";
-        private const string AutoLogFileName = "{AutoLogFileName}";
-        private readonly string AutoLogFileNameReplacement = string.Format("{0}_{1}_{2}_{3}", AssemblyName, Date, Time, ProcessId);
+        /// <summary>
+        /// Defines the keywords that can be used in the <see cref="FilePath"/> property to inject different values.
+        /// </summary>
+        public static class FilePathKeyword
+        {
+            /// <summary>
+            /// Keyword that can be used within the <see cref="FilePath"/> to create a log with the following format: 
+            /// <para/>
+            /// <c>{AssemblyName}_{Date}_{Time}_{ProcessId}</c>
+            /// </summary>
+            public const string AutoLogFileName = "{AutoLogFileName}";
+
+            /// <summary>
+            /// Keyword that can be used within the <see cref="FilePath"/> to insert <c>%ProgramData%</c> or <c>%AppData%</c> into the filepath.
+            /// <para/>
+            /// If the application is a web app (System.Web is referenced); then <c>%ProgramData%</c> is used; otherwise <c>%AppData%</c> is used.
+            /// </summary>
+            public const string AppData = "{AppData}";
+
+            /// <summary>
+            /// Keyword that can be used within the <see cref="FilePath"/> to insert <c>%LocalAppData%</c> into the filepath.
+            /// </summary>
+            public const string AppDataLocal = "{AppDataLocal}";
+
+            /// <summary>
+            /// Keyword that can be used within the <see cref="FilePath"/> to insert <c>%AppData%</c> into the filepath.
+            /// </summary>
+            public const string AppDataRoaming = "{AppDataRoaming}";
+
+            /// <summary>
+            /// Keyword that can be used within the <see cref="FilePath"/> to insert <c>%ProgramData%</c> into the filepath.
+            /// </summary>
+            public const string AppDataMachine = "{AppDataMachine}";
+
+            /// <summary>
+            /// Keyword that can be used within the <see cref="FilePath"/> to insert the <c>AppDomain.CurrentDomain.BaseDirectory</c> into the filepath.
+            /// </summary>
+            public const string AppDir = "{AppDir}";
```

Comment:
I think _App_ would be more consistent. But I can imagine it would be a bit confusing because we also have AssemblyName, etc.